src/Extensions/Diagnostics.HealthChecks/Internal/ServiceContextHealthStatusSender.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Fabric;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.Extensions.Abstractions;
+using System.Fabric.Health;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	internal class ServiceContextHealthStatusSender : IHealthStatusSender
+	{
+		private const string HealthReportSourceId = nameof(ServiceContextHealthStatusSender);
+
+		private readonly IAccessor<IServicePartition> m_partitionAccessor;
+
+		private readonly ILogger<ServiceContextHealthStatusSender> m_logger;
+
+		private Action<HealthInformation>? m_reportHealth;
+
+		public ServiceContextHealthStatusSender(IAccessor<IServicePartition> partitionAccessor, ILogger<ServiceContextHealthStatusSender> logger)
+		{
+			m_partitionAccessor = partitionAccessor;
+			m_logger = logger;
+		}
+
+		public Task IntializeAsync(CancellationToken token)
+		{
+			IServicePartition? partition = m_partitionAccessor.Value;
+			if (partition == null)
+			{
+				m_logger.LogWarning(Tag.Create(), "Publisher run before partition provided.");
+			}
+			else
+			{
+				// Avoiding repeated pattern matching for each report entry.
+				m_reportHealth = partition switch
+				{
+					IStatefulServicePartition statefulPartition => statefulPartition.ReportReplicaHealth,
+					IStatelessServicePartition statelessPartition => statelessPartition.ReportInstanceHealth,
+					_ => throw new ArgumentException($"Service partition type '{partition.GetType()}' is not supported."),
+				};
+			}
+
+			return Task.CompletedTask;
+		}
+
+		public Task SendStatusAsync(string checkName, HealthStatus status, string description, CancellationToken token)
+		{
+			if (m_reportHealth == null)
+			{
+				m_logger.LogWarning(Tag.Create(), "Publisher run before health report functions is provided.");
```

Comment:
I'm not clear on what this message is trying to convey.
Should it be "The status sender was executed before a report was generated." or something like that?

Reply:
Updated message and logic to throw exception instead